packages/typography/src/elements/CodeBlock.tsx

```diff
@@ -18,13 +24,37 @@ export interface ICodeBlockProps extends HTMLAttributes<HTMLPreElement> {
   isLight?: boolean;
   /** Displays line numbers */
   isNumbered?: boolean;
+  /** Determines the lines to highlight */
+  highlightLines?: number[];
+  /** Passes props to the code block container */
+  containerProps?: HTMLAttributes<HTMLDivElement>;
 }
 
 /**
  * @extends HTMLAttributes<HTMLPreElement>
  */
 export const CodeBlock = React.forwardRef<HTMLPreElement, ICodeBlockProps>(
-  ({ children, isLight, isNumbered, language, size, ...other }, ref) => {
+  (
+    { children, containerProps, highlightLines, isLight, isNumbered, language, size, ...other },
+    ref
+  ) => {
+    const [containerTabIndex, setContainerTabIndex] = useState(-1);
+    const containerRef = useRef<HTMLDivElement>(null);
+    const updateContainerTabIndex = debounce(() => {
```

Comment:
With this no longer being memoized, would a re-render (not including props that are dependencies within the useEffect) remove the correct listener?

Even with no dependencies I think memoizing this is necessary. This would create a new debounce reference on every render.